flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -473,8 +474,13 @@ private ContainerLaunchContext createTaskExecutorLaunchContext(Resource resource
 
 		log.debug("TaskManager configuration: {}", flinkConfig);
 
+		Configuration taskManagerConfig = flinkConfig.clone();
+		if (!flinkConfig.containsKey(CoreOptions.TMP_DIRS_OVERRIDDEN)){
+			taskManagerConfig.setString(CoreOptions.TMP_DIRS, "");	// HACK: emulate removal for the given key
+		}
+
 		ContainerLaunchContext taskExecutorLaunchContext = Utils.createTaskExecutorContext(
-			flinkConfig,
```

Comment:
here we should keep the `flinkConfig` and instead replace the 5th argument with `taskManagerConfig` because it is this `Configuration` which is given to the new `TaskManager` container.

Reply:
done. Thank you very much!
Since you seem fine with this hacky approach I have tested and made stable the last PR on our Yarn cluster.